components/eventing-controller/api/v1alpha1/subscription_lifecycle.go

```diff
@@ -76,6 +76,24 @@ func (s SubscriptionStatus) IsReady() bool {
 	return true
 }
 
+func (s SubscriptionStatus) FindCondition(conditionType ConditionType) *Condition {
+	for _, condition := range s.Conditions {
+		if conditionType == condition.Type {
+			return &condition
+		}
+	}
+	return nil
+}
+
+// ShouldUpdateReadyStatus checks if there is a mismatch between the
+// subscription Ready Status and the Ready status of all the conditions
+func (s SubscriptionStatus) ShouldUpdateReadyStatus() bool {
```

Comment:
Should we have some unit tests for these two new methods? At least for `ShouldUpdateReadyStatus`?

Reply:
done